src/Debugging-Core-Tests/DebuggingInstructionStreamTest.class.st

```diff
@@ -0,0 +1,141 @@
+Class {
+	#name : #DebuggingInstructionStreamTest,
+	#superclass : #TestCase,
+	#category : #'Debugging-Core-Tests'
+}
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> method1 [
+
+	| a |
+	a := 1.
+	^ 1 + 2
+]
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> methodWithBlock [
+
+	[| a |
+	a := 1.
+	^ 1 + 2] value
+]
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> methodWithSuper [
+
+	super printString.
+	^ 42
+]
+
+{ #category : #'tests - structure' }
+DebuggingInstructionStreamTest >> testFirstByte [
+	|compiledMethod stream|
+	compiledMethod := self class >> #method1.
+	stream := InstructionStream on: compiledMethod.
+	self assert: stream firstByte equals: (compiledMethod at: compiledMethod initialPC)
+]
+
+{ #category : #'tests - structure' }
+DebuggingInstructionStreamTest >> testFollowingByteAccessByteCodeConstituentByteByByte [
+
+	| compiledMethod stream client |
+	compiledMethod := self class >> #methodWithSuper.
+	stream := InstructionStream on: compiledMethod.
+	client := InstructionClient new.
+	
+	"Answer the next byte.
+	For example for this bytecode 
+	<EB 00> superSend 
+	<D8> pop
+	 If the current byte is EB, the following byte is 00.
+	 It is not the following bytecode, that is D8"
+ 	stream pc: stream pc + 1.	
+	self assert: stream followingByte identicalTo: (compiledMethod at: compiledMethod initialPC + 2).
+	self deny: stream followingByte identicalTo: stream followingBytecode.
+	stream pc: stream pc + 1.
+	self assert: stream followingByte identicalTo: (compiledMethod at: compiledMethod initialPC + 3).
+	stream pc: stream pc + 1.
+	self assert: stream followingByte identicalTo: stream followingBytecode
+]
+
+{ #category : #'tests - structure' }
+DebuggingInstructionStreamTest >> testFollowingByteWithSuper [
+
+	| compiledMethod stream client |
+	compiledMethod := self class >> #methodWithSuper.
+	stream := InstructionStream on: compiledMethod.
+	client := InstructionClient new.
+	stream interpretNextInstructionFor: client.
+	self flag: 'improve this test to avoid magic number'.
+	self
+		assert: stream followingByte
+		identicalTo: (compiledMethod at: 43)
+]
+
+{ #category : #'tests - interpretation' }
+DebuggingInstructionStreamTest >> testInterpretNextInstruction [
+	| compiledMethod stream client currentPC |
+	compiledMethod := self class >> #method1.
+	stream := InstructionStream on: compiledMethod.
+	client := InstructionClient new.
+	currentPC := stream pc.
+	stream interpretNextInstructionFor: client.
```

Comment:
This is to me a too generic name for the test. It really depends on what bytecode you are...

Reply:
agreed